src/Database/Table/GroupedSelection.php

```diff
@@ -201,6 +201,10 @@ protected function loadRefCache()
 		}
 	}
 
+	protected function emptyResultSet($saveCache = TRUE, $deleteRererencedCache = TRUE) {
+		parent::emptyResultSet($saveCache, false);
```

Comment:
Difference:

1) My solution:
When `emptyResultSet()` is called on Selection, the `$this->refCache['referenced'] = [];` line is executed. (the default value of second parameter is true)

When `emptyResultSet()` is called on GroupedSelection, the `$this->refCache['referenced'] = [];` is not executed (the second parameter of emptyResultSet is false)

2) Your solution
Its actually the opposite version of mine.